src/radical/pilot/agent/resource_manager/base.py

```diff
@@ -320,31 +320,27 @@ def _prepare_launch_methods(self, rm_info):
         launch_methods  = self._cfg.resource_cfg.launch_methods
 
         self._launchers    = {}
-        self._launch_order = None
+        self._launch_order = launch_methods.get('order') or list(launch_methods)
```

Comment:
I didn't even know that `list(some_dict)` gives  a list of keys - always used `list(some_dict.keys())` for that (which removes the iterator).  Thanks for that tidbit :-)  We should clean that up throughout the code when we stumble over it...